### PR TITLE
Added back the reset function

### PR DESCRIPTION
**Pull request by eduardonunesp**
- Small refactoring on eth subscription structure

### CODE REVIEW

File: src/loom-provider.ts [hunk 1]
```diff
@@ -383,6 +372,10 @@ export class LoomProvider {
     this._client.disconnect()
   }
 
+  reset() {
```
Comment (enlight) [inline src/loom-provider.ts]: This one line function is still as pointless as when I removed it, why did you add it back?

File: src/loom-provider.ts [hunk 2]
```diff
@@ -171,8 +178,10 @@ export class LoomProvider {
         return createDefaultTxMiddleware(client, privateKey)
       }
     }
+    this._addDefaultMethods()
```
Comment (enlight) [inline src/loom-provider.ts]: How is this supposed to work exactly? When the client is setup to use the `/eth` endpoint the `eth_XXX` methods should be handled by `_ethCallSupportedMethod` which skips the protobuf marshalling & unmarshalling. This change here prevents that from happening, and send protobufs to the `/eth` endpoint which doesn't know what to do with them.

File: src/loom-provider.ts (hunk 1 above)
Comment (eduardonunesp) [reply src/loom-provider.ts]: Reset function isn't pointless is used internally by the `Web3` to reset the callbacks if necessary by the request manager `https://github.com/ethereum/web3.js/blob/1.x/packages/web3-core-requestmanager/src/index.js#L227-L228`

File: src/loom-provider.ts (hunk 2 above)
Comment (eduardonunesp) [reply src/loom-provider.ts]: My bad this one will be fixed

File: src/loom-provider.ts (hunk 1 above)
Comment (enlight) [reply src/loom-provider.ts]: OK, thanks for the clarification, I've added a comment to ensure it's not removed again in the future.
